File: apps/android/app/src/main/java/com/notiving/notiving/shell/router/ShellRouter.kt
package com.notiving.notiving.shell.router

import android.content.Context
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow

class ShellRouter(context: Context) {
    val config: RouteTableConfig = RouteTableLoader.load(context)

    private val _selectedTab = MutableStateFlow(config.tabs.first().key)
    val selectedTab: StateFlow<String> = _selectedTab

    fun resolve(path: String): RouteEntry? =
        config.routes.firstOrNull { it.pattern == path }

    fun navigate(url: String) {
        val tab = config.tabs.firstOrNull { it.route == url }
        if (tab != null) {
            _selectedTab.value = tab.key
        }
    }

    fun back() {
        // Tab-level: no-op for now. Push navigation in Phase 4.
    }

    fun setTitle(title: String) {
        // Phase 4: update native nav bar title
    }

    fun setNavBarHidden(hidden: Boolean) {
        // Phase 4: show/hide native nav bar
    }

    companion object {
        var instance: ShellRouter? = null
            private set

        fun init(context: Context): ShellRouter {
            val router = ShellRouter(context)
            instance = router
            return router
        }
    }
}

File: apps/ios/notiving/notiving/Shell/Router/ShellRouter.swift
import Combine
import Foundation

final class ShellRouter: ObservableObject {
    static let shared = ShellRouter()

    let config: RouteTableConfig
    @Published var selectedTab: String

    init() {
        self.config = RouteTableLoader.load()
        self.selectedTab = config.tabs.first?.key ?? "home"
    }

    func resolve(_ path: String) -> RouteEntry? {
        config.routes.first { $0.pattern == path }
    }

    func navigate(to url: String) {
        if let tab = config.tabs.first(where: { $0.route == url }) {
            selectedTab = tab.key
        }
    }

    func back() {
        // Tab-level: no-op for now. Push navigation in Phase 4.
    }

    func setTitle(_ title: String) {
        // Phase 4: update native nav bar title
    }

    func setNavBarHidden(_ hidden: Bool) {
        // Phase 4: show/hide native nav bar
    }
}
